krun/results.py

```diff
@@ -92,6 +99,15 @@ def write_to_file(self):
             f.write(json.dumps(to_write,
                                indent=1, sort_keys=True, encoding='utf-8'))
 
+    def add_instrumentation_data(self, bench_key, inst_dct):
+        """Record instrumentation data into results object."""
+
+        for inst_key, v in inst_dct.iteritems():
+            # keys in dict are arbitrary, so we may need to scaffold a new list
```

Comment:
The keys that appear in the instrumentation data are not known ahead of time. This means different VMs can provide different kinds of data if we so wish. Upshot: we have to instantiate the keys in the results file in an ad-hoc "when we see them" fashion. I will update the comment.